### PR TITLE
Create an `AnalysisPromptCategory` entity

### PR DESCRIPTION
**Pull request by isxoqovxasanboy**

### CODE REVIEW

**Review by rashidovasadbek — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/PromptCategory/Models/AnalysisPromptCategoryDto.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    
+    public FeedbackAnalysisPromptCategory Category { get; set; }
+
+    public string TypeDisplayName => Category.GetDisplayName(); 
+    
+    public int PromptsCount { get; set; }
+    
+    public Guid? SelectedPromptId { get; set; }
+    
+    public AnalysisPromptDto? SelectedPrompt { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/PromptCategory/Models/AnalysisPromptCategoryDto.cs:19]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/PromptCategory/Models/PromptCategoryFilter.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+using Feedback.Analyzer.Domain.Common.Query;
+
+namespace Feedback.Analyzer.Application.Common.PromptCategory.Models;
+
+public class PromptCategoryFilter : FilterPagination
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/PromptCategory/Models/PromptCategoryFilter.cs:5]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/PromptCategory/Services/IPromptCategoryService.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+{
+    IQueryable<AnalysisPromptCategory> Get(
+        Expression<Func<AnalysisPromptCategory, bool>>? predicate = default,
+        QueryOptions queryOptions = default
+    );
+
+    IQueryable<AnalysisPromptCategory> Get(
+        PromptCategoryFilter promptCategoryFilter, 
+        QueryOptions queryOptions = default);
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/PromptCategory/Services/IPromptCategoryService.cs:18]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Models/AnalysisPromptDto.cs [hunk 4]
```diff
@@ -7,43 +7,19 @@ namespace Feedback.Analyzer.Application.Common.Prompts.Models;
...
-    /// <summary>
-    /// Gets or sets the creation time of the analysis prompt.
-    /// </summary>
     public DateTimeOffset CreatedTime { get; set; }
     
-    /// <summary>
-    /// Gets or sets the modified time of the analysis prompt.
-    /// </summary>
     public DateTimeOffset ModifiedTime { get; set; }
 }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Models/AnalysisPromptDto.cs:25]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Models/PromptResultDto.cs [hunk 5]
```diff
@@ -28,10 +28,11 @@ public record PromptResultDto
     /// <summary>
     /// Gets or sets the average execution time for the prompt.
     /// </summary>
-    public TimeSpan AverageExecutionTime { get; set; }
+    public int AverageExecutionTimeInSeconds { get; set; }
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Models/PromptResultDto.cs:31]: Change to milliseconds

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Queries/PromptGetByIdQuery.cs [hunk 6]
```diff
@@ -11,5 +11,5 @@ public record PromptGetByIdQuery : IQuery<AnalysisPromptDto?>
     /// <summary>
     /// The unique identifier of the organization to retrieve.
     /// </summary>
-    public Guid PromptId { get; set; } = default!;
+    public Guid PromptId { get; set; } = Guid.Empty;
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Application/Common/Prompts/Queries/PromptGetByIdQuery.cs:14]: Remove Guid.Empty

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/Prompts/QueryHandlers/PromptResultGetByCategoryIdQueryHandler.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+using Feedback.Analyzer.Application.Common.Prompts.Models;
+using Feedback.Analyzer.Application.Common.Prompts.Queries;
+using Feedback.Analyzer.Application.Common.Prompts.Services;
+using Feedback.Analyzer.Domain.Common.Queries;
+using Microsoft.EntityFrameworkCore;
+
+namespace Feedback.Analyzer.Infrastructure.Common.Prompts.QueryHandlers;
+
+public class PromptResultGetByCategoryIdQueryHandler(IMapper mapper, IPromptService promptService) 
+    : IQueryHandler<PromptResultGetByCategoryIdQuery, ICollection<PromptResultDto>>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/Prompts/QueryHandlers/PromptResultGetByCategoryIdQueryHandler.cs:11]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/PromptsCategories/QueryHandlers/PromptCategoryGetQueryHandler.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+using Feedback.Analyzer.Application.Common.PromptCategory.Models;
+using Feedback.Analyzer.Application.Common.PromptCategory.Quearies;
+using Feedback.Analyzer.Application.Common.PromptCategory.Services;
+using Feedback.Analyzer.Domain.Common.Queries;
+using Microsoft.EntityFrameworkCore;
+
+namespace Feedback.Analyzer.Infrastructure.Common.PromptsCategories.QueryHandlers;
+
+public class PromptCategoryGetQueryHandler(IMapper mapper, IPromptCategoryService promptCategoryService)
+    : IQueryHandler<PromptCategoryGetQuery, ICollection<AnalysisPromptCategoryDto>>
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/PromptsCategories/QueryHandlers/PromptCategoryGetQueryHandler.cs:12]: Add docs

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/PromptsCategories/QueryHandlers/PromptCategoryGetQueryHandler.cs [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    {
+        var matchedPromptCategory = await promptCategoryService.Get(
+            request.Filter,
+            new QueryOptions
+            {
+                AsNoTracking = true
+            }
+        )
+        .Include(promptCategory => promptCategory.Prompts)
+        //.ProjectTo<AnalysisPromptCategoryDto>(mapper.ConfigurationProvider)
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/PromptsCategories/QueryHandlers/PromptCategoryGetQueryHandler.cs:24]: Remove commented code

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/PromptsCategories/Services/PromptCategoryService.cs [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+using Feedback.Analyzer.Domain.Common.Queries;
+using Feedback.Analyzer.Domain.Entities;
+using Feedback.Analyzer.Persistence.Extensions;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+using FluentValidation;
+
+namespace Feedback.Analyzer.Infrastructure.Common.PromptsCategories.Services;
+
+public class PromptCategoryService(IPromptCategoryRepository promptCategoryRepository, IValidator<AnalysisPrompt> promptValidator)
+    : IPromptCategoryService
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/PromptsCategories/Services/PromptCategoryService.cs:13]: Add docs

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by bunyodkamalov — APPROVED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/PromptsCategories/Services/PromptCategoryService.cs [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+using Feedback.Analyzer.Persistence.Extensions;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+using FluentValidation;
+
+namespace Feedback.Analyzer.Infrastructure.Common.PromptsCategories.Services;
+
+/// <summary>
+/// Represents a service that provides operations for prompt categories.
+/// </summary>
+public class PromptCategoryService(IPromptCategoryRepository promptCategoryRepository, IValidator<AnalysisPrompt> promptValidator)
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Infrastructure/Common/PromptsCategories/Services/PromptCategoryService.cs:15]: Remove validator

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by sulton-max — APPROVED** [review]

**Review by bunyodkamalov — APPROVED** [review]

**Pull request merged**